public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts

```diff
@@ -98,23 +97,7 @@ export class CloudWatchLogsQueryRunner extends CloudWatchRequest {
       ),
     }));
 
-    const hasQueryWithMissingLogGroupSelection = queryParams.some((qp) => {
-      const missingLogGroupNames = qp.logGroupNames.length === 0;
-      const missingLogGroups = qp.logGroups.length === 0;
-      return missingLogGroupNames && missingLogGroups;
-    });
-
-    if (hasQueryWithMissingLogGroupSelection) {
-      return of({ data: [], error: { message: 'Log group is required' } });
-    }
-
-    const hasQueryWithMissingQueryString = queryParams.some((qp) => {
-      return qp.queryString.length === 0;
-    });
-
-    if (hasQueryWithMissingQueryString) {
-      return of({ data: [], error: { message: 'Query is required' } });
-    }
+    const validLogQueries = startQueryRequests.filter(this.filterQuery);
```

Comment:
Can the filtering be moved so that it happens before creating the `startQueryRequests` array?